src/bitcoin/mod.rs

```diff
@@ -95,10 +98,7 @@ pub trait BitcoinInterface: Send {
     fn spent_coins(
         &self,
         outpoints: &[(bitcoin::OutPoint, bitcoin::Txid)],
-    ) -> (
-        Vec<(bitcoin::OutPoint, bitcoin::Txid, Block)>,
-        Vec<bitcoin::OutPoint>,
-    );
+    ) -> (Vec<SpentCoin>, Vec<bitcoin::OutPoint>);
```

Comment:
I should have added more comments :slightly_smiling_face: 

It was somehow easier to get the time and height separately rather than as part of `Block` (possibly because I didn't have  the hash readily available in the Electrum part and this was required to define `Block`).

Then I named the type to avoid a clippy warning.

It may make sense to refactor this, but at the time I was focused on other aspects and wanted to make the quickest change possible :face_in_clouds: